Author: Luciano Bello <luciano@linux.org.ar>
Description: Avoid the "implicit declaration of function 'str*'" warning
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/

--- a/arp.c
+++ b/arp.c
@@ -34,6 +34,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <string.h>
 
 #include "arp.h"
 
--- a/buf.c
+++ b/buf.c
@@ -17,6 +17,7 @@
 #include <unistd.h>
 #include <ctype.h>
 #include <err.h>
+#include <string.h>
 
 #include "buf.h"
 
--- a/decode_nntp.c
+++ b/decode_nntp.c
@@ -15,6 +15,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "base64.h"
 #include "decode.h"
--- a/decode_pop.c
+++ b/decode_pop.c
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "base64.h"
 #include "options.h"
--- a/decode_rlogin.c
+++ b/decode_rlogin.c
@@ -14,6 +14,8 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcpy.h>
+#include <strlcat.h>
 
 #include "options.h"
 #include "decode.h"
--- a/decode_smb.c
+++ b/decode_smb.c
@@ -15,6 +15,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "decode.h"
 
--- a/decode_smtp.c
+++ b/decode_smtp.c
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "base64.h"
 #include "options.h"
--- a/decode_sniffer.c
+++ b/decode_sniffer.c
@@ -15,6 +15,8 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
+#include <strlcpy.h>
 
 #include "base64.h"
 #include "decode.h"
--- a/decode_socks.c
+++ b/decode_socks.c
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "decode.h"
 
--- a/decode_tds.c
+++ b/decode_tds.c
@@ -18,6 +18,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "decode.h"
 
--- a/decode_telnet.c
+++ b/decode_telnet.c
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcpy.h>
 
 #include "options.h"
 #include "decode.h"
--- a/decode_x11.c
+++ b/decode_x11.c
@@ -14,6 +14,8 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
+#include <strlcpy.h>
 
 #include "decode.h"
 
--- a/dnsspoof.c
+++ b/dnsspoof.c
@@ -20,6 +20,7 @@
 #include <stdlib.h>
 #include <signal.h>
 #include <string.h>
+#include <strlcpy.h>
 #include <resolv.h>
 #include <err.h>
 #include <libnet.h>
--- a/magic.c
+++ b/magic.c
@@ -36,6 +36,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <strlcpy.h>
 #include <ctype.h>
 #include <time.h>
 #include <err.h>
--- /dev/null
+++ b/missing/strlcat.h
@@ -0,0 +1 @@
+size_t strlcat(char *dst, const char *src, size_t siz);
--- /dev/null
+++ b/missing/strlcpy.h
@@ -0,0 +1 @@
+size_t strlcpy(char *dst, const char *src, size_t siz);
--- a/sshmitm.c
+++ b/sshmitm.c
@@ -24,6 +24,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <strlcat.h>
 
 #include "buf.h"
 #include "record.h"
